Makefile

```diff
@@ -95,9 +108,22 @@ vet:
 	go vet ./...
 
 # Generate code
-generate: controller-gen
+generate: controller-gen types
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
 
+# generates many other files (listers, informers, client etc).
+api/addon/v1alpha1/zz_generated.deepcopy.go: $(TYPES)
```

Comment:
Never seen this style of make target what's it doing?

Reply:
Similar usage 
stuff https://github.com/argoproj/argo-workflows/blob/master/Makefile#L251
https://github.com/argoproj/argo-workflows/blob/master/Makefile#L530

Just need keep the go file updated

Reply:
I still don't understand this usage, we don't need a symlink at all in the codebase, if the deepcopy is generated by the tool now we should just delete this file and we should remove the deepcopy comment markers.

Reply:
The symlink is just ensure the file get updated successfully, no error, the build remove it in the end https://github.com/keikoproj/addon-manager/pull/116/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R122

Every API needs the zz_generated.deepcopy.go file, otherwise, object instance deepCopy fails, controller could not reconcile.

Reply:
The `code-gen` target should be generating the `zz_generated.deepcopy.go` file, is that still the case or is it being modified by the below command?

Reply:
controller-gen should be generating zz_generated.deepcopy.go. and the code-generator which use deepcopy-gen, which is part of controller-gen update the zz_generated.deepcopy.go file when generating clientset/informer/listers. 
The only difference between is code-generator has strict syntax , such as return statement fix, otherwise, everything else is the same.